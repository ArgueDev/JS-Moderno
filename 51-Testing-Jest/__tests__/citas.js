import Citas from '../js/classes/Citas.js';

describe('Probar la clse de citas', () => {
    
    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Christian',
            telefono: '0969536446',
            fecha: '10-12-2024',
            hora: '10:03',
            sintomas: 'Solo duerme'
        };

        citas.agregarCita(citaObj);

        // Prueba 
        expect(citas).toMatchSnapshot();

    });

    // Actualizar cita
    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Peluchin',
            propietario: 'Christian',
            telefono: '0969536446',
            fecha: '10-12-2024',
            hora: '10:03',
            sintomas: 'Solo duerme'
        };

        citas.editarCita(citaActualizada);

        // Prueba
        expect(citas).toMatchSnapshot();
    });

    // Eliminar cita
    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        // Prueba
        expect(citas).toMatchSnapshot();
    })
});