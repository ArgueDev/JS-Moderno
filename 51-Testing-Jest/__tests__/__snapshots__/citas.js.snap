// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clse de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2024",
      "hora": "10:03",
      "id": 1733152968429,
      "mascota": "Peluchin",
      "propietario": "Christian",
      "sintomas": "Solo duerme",
      "telefono": "0969536446",
    },
  ],
}
`;

exports[`Probar la clse de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2024",
      "hora": "10:03",
      "id": 1733152968429,
      "mascota": "Hook",
      "propietario": "Christian",
      "sintomas": "Solo duerme",
      "telefono": "0969536446",
    },
  ],
}
`;

exports[`Probar la clse de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
